restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java

```diff
@@ -208,6 +144,133 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
     }
   }
 
+  private ImmutablePair<Map<String, Object>, List<String>> convertJsonNodeToBoundVariables(
+      JsonNode jsonValue,
+      ParsingContext parsingContext,
+      DocumentDB db,
+      List<String> path,
+      String key,
+      boolean patching) {
+    List<String> firstLevelKeys = new ArrayList<>();
```

Comment:
can you just move this closer to it's usage.. It's so hard to read this code, if you initialize a var at like `154` and first time use it on line `184`..

Would also be great to add small comment to each private method what's doing and small inline comments in the bigger methods, so somebody can follow the implementation.. It's so hard to read..